docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
+
+A detailed list of the subcommand classes is as follows:
+* Project Commands:  
+    * `CreateProjectCommand`
+    * `ListProjectCommand`
+    * `SelectProjectCommand`
+    * `ViewProjectCommand`
+* Member Commands:
+    * `AddMemberCommand`
+    * `DeleteMemberCommand`
+* Task Commands:
+    * `AddTaskCommand`
+    * `ChangeTaskPriorityCommand`
+    * `DeleteTaskCommand`
+    * `DoneTaskCommand`
+    * `ViewTaskCommand`
+* Sprint Commands:
+    * `AddSprintTaskCommand`
+    * `AllocateSprintTaskCommand`
+    * `CreateSprintCommand`
+    * `DeallocateSprintTaskCommand`
+    * `EditSprintCommand`
+    * `RemoveSprintTaskCommand`
+    * `ViewSprintCommand`
+* Help Commands:
+    * `HelpCommand`
+    * `MemberHelpCommand`
+    * `ProjectHelpCommand`
+    * `SprintHelpCommand`
+    * `StorageHelpCommand`
+    * `TaskHelpCommand`
+### Model Component
+![Figure X: Simplified class diagram for Model Component](./image/developerguide/modelcomponent.png "Storage Component UML")  
+Link: [Model Package](https://github.com/AY2021S1-CS2113T-F11-4/tp/tree/master/src/main/java/seedu/duke/model)  
+The Model
+* Includes four packages namely, Project, Task, Member and Sprint.
+
+Project package
+* Includes a ProjectList to manage the multiple instances of Project created by the user
+* Each instance of Project stores one instance of ProjectBacklog, ProjectMembers and SprintList.
+
+Task  package
+* Includes a ProjectBacklog to manage every Tasks created by the user
+* Task can be allocated to Sprints and can be assigned to Members, 
+
+Member package
+* Includes a ProjectMembers to manage every Members created by the user
+* Member can be assigned with Tasks and can be allocated to Sprints holding those Tasks
+ 
+Sprint package
+* Includes a SprintList to manage every Sprints created by the user
+* Sprint can contain Tasks and Members allocated to those Tasks
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
+  
+API: [StorageManager.java](/src/main/java/seedu/duke/storage/StorageManager.java)  
+The Storage component is using the JavaScript Object Notation (JSON) to save the data. The library used for serialising and deserializing the data is _json.simple 3.1.1_ by **Clifton Labs**.  
+As shown in the diagram above, `JsonableObject` and `JsonableArray` are interfaces which inherits the `Jsonable` interface. The following model class inherits only one of the two interfaces:  
+- ProjectManager  
+- Project  
+- SprintManager  
+- TaskManager  
+- ProjectMembers  
+- Sprint  
+- Task  
+- Member  
+
+This requires the model classes to implement two methods required for JSON serialisation and deserialisation:  
+- `toJson()`: Contains logic required to convert the model object into JSON string.  
+- `fromJson()`: Contains logic required to convert JSON object into its respective model class.    
+
+## Implementation
```

Comment:
Perhaps there could be prefaces for these headers? Maybe even a diagram as well?